File: src/main/java/com/aquasecurity/plugins/trivy/ui/treenodes/MisconfigurationTreeNode.java
package com.aquasecurity.plugins.trivy.ui.treenodes;

import com.aquasecurity.plugins.trivy.icons.TrivyIcons;
import com.aquasecurity.plugins.trivy.model.Misconfiguration;

import javax.swing.*;
import javax.swing.tree.DefaultMutableTreeNode;

class MisconfigurationTreeNode extends DefaultMutableTreeNode implements TrivyTreeNode {

    public final Misconfiguration misconfiguration;

    public MisconfigurationTreeNode(Misconfiguration misconfiguration) {
        super(misconfiguration);
        this.misconfiguration = misconfiguration;
    }

    @Override
    public Icon getIcon() {
        return TrivyIcons.Trivy;
    }

    @Override
    public String getTitle() {
        return this.misconfiguration.id;
    }

    @Override
    public String getTooltip() {
        return this.misconfiguration.description;
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/ui/treenodes/MisconfigurationTreeNode.kt
package com.aquasecurity.plugins.trivy.ui.treenodes

import com.aquasecurity.plugins.trivy.icons.TrivyIcons
import com.aquasecurity.plugins.trivy.model.Misconfiguration
import javax.swing.Icon
import javax.swing.tree.DefaultMutableTreeNode

internal class MisconfigurationTreeNode(val misconfiguration: Misconfiguration) :
    DefaultMutableTreeNode(misconfiguration), TrivyTreeNode {
    override val icon: Icon?
        get() = TrivyIcons.Trivy
    override val title: String?
        get() = misconfiguration.id!!
    override val tooltip: String?
        get() = misconfiguration.description!!
}